### DISCUSSION TITLE
Missing setuptools when using virtual environments in Pycharm

### DISCUSSION THREAD

**Discussion by wu-clan** [pdm-project/pdm#2448] [Q&A]
When starting a new project in Pycharm 2023.2.5, the installer does not get the smart import prompt and a popup window displays `Error: Python packaging tool 'setuptools' not found` when you look at the virtual environment's installed packages in Settings.

Is this a Pycharm version problem or am I missing something?

**Comment by wu-clan** [comment]
When I `pip install setuptools`, Pycharm will function correctly, the reason seems to be that pycharm can't parse the packages in the environment without setuptools.
![image](https://github.com/pdm-project/pdm/assets/52145145/627cdd54-3e90-4c02-aa49-34d5d94b6e7d)

**Comment by wu-clan** [comment]
Well, even though I have setuptools installed, I still can't get the smart import prompt for installed packages

**Comment by wu-clan** [comment]
In hindsight, I didn't do anything, but it magically allows for intelligent prompting

**Reply by frostming** [reply]
Perhaps it takes time for Pycharm to seed the packages.
